arrow.go

```diff
@@ -0,0 +1,467 @@
+			}
+		}
+		arr, err = bld.Finish()
+		if err != nil {
+			return nil, errors.Wrap(err, "timestamp: can't create array")
+		}
+	default:
+		return nil, fmt.Errorf("unkown data type - %T", data)
+	}
+
+	col, err := carrow.NewColumn(field, arr)
+	if err != nil {
+		return nil, errors.Wrap(err, "can't create column")
+	}
+
+	return &ArrowColumn{col}, nil
+}
+
+// ArrowColumnBuilder builds arrow based columns
+type ArrowColumnBuilder struct {
+	field        *carrow.Field
+	boolBuilder  *carrow.BoolArrayBuilder
+	floatBuilder *carrow.Float64ArrayBuilder
+	intBuilder   *carrow.Int64ArrayBuilder
+	strBuilder   *carrow.StringArrayBuilder
+	tsBuilder    *carrow.TimestampArrayBuilder
+}
+
+// NewArrowColumnBuilder return new ArrowColumnBuilder
+func NewArrowColumnBuilder(name string, dtype DType, size int) (*ArrowColumnBuilder, error) {
+	var typ carrow.DType
+	bld := &ArrowColumnBuilder{}
+	switch dtype {
+	case BoolType:
+		typ = carrow.BoolType
+		bld.boolBuilder = carrow.NewBoolArrayBuilder()
+	case FloatType:
+		typ = carrow.Float64Type
+		bld.floatBuilder = carrow.NewFloat64ArrayBuilder()
+	case IntType:
+		typ = carrow.Integer64Type
+		bld.intBuilder = carrow.NewInt64ArrayBuilder()
+	case StringType:
+		typ = carrow.StringType
+		bld.strBuilder = carrow.NewStringArrayBuilder()
+	case TimeType:
+		typ = carrow.TimestampType
+		bld.tsBuilder = carrow.NewTimestampArrayBuilder()
+	default:
+		return nil, fmt.Errorf("unsupported dtype - %s", dtype)
+	}
+
+	var err error
+	bld.field, err = carrow.NewField(name, typ)
+	if err != nil {
+		return nil, err
+	}
+
+	return bld, nil
+}
+
+// Append appends a value
+func (b *ArrowColumnBuilder) Append(value interface{}) error {
+	switch b.field.DType() {
+	case carrow.BoolType:
+		bval, ok := value.(bool)
+		if !ok {
+			return typeError(value, "bool")
+		}
+		b.boolBuilder.Append(bval)
+	case carrow.Float64Type:
+		fval, err := asFloat64(value)
+		if err != nil {
+			return err
+		}
+		return b.floatBuilder.Append(fval)
+	case carrow.Integer64Type:
+		ival, err := asInt64(value)
+		if err != nil {
+			return err
+		}
+		return b.intBuilder.Append(ival)
+	case carrow.StringType:
+		sval, ok := value.(string)
+		if !ok {
+			return typeError(value, "string")
+		}
+		return b.strBuilder.Append(sval)
+	case carrow.TimestampType:
+		tval, ok := value.(time.Time)
+		if !ok {
+			return typeError(value, "time.Time")
+		}
+		return b.tsBuilder.Append(tval)
+	}
+	return fmt.Errorf("unsupported dtype - %s", b.field.DType())
+}
+
+// At return value at
+func (b *ArrowColumnBuilder) At(index int) (interface{}, error) {
+	return nil, fmt.Errorf("not implemented")
+}
+
+// Set sets a value
+func (b *ArrowColumnBuilder) Set(index int, value interface{}) error {
+	return fmt.Errorf("not implemented")
+}
+
+// Delete deletes a value
+func (b *ArrowColumnBuilder) Delete(index int) error {
+	return fmt.Errorf("not implemented")
+}
+
+// Finish create the colum
+func (b *ArrowColumnBuilder) Finish() Column {
```

Comment:
Usually this is called `Build` or `Get`

Reply:
Agree. However I'm trying to keep the same name [Arrow C++ uses](https://arrow.apache.org/docs/cpp/classarrow_1_1_array_builder.html#a67b2227edb4b51bc44425f1c8a439819)